### PR TITLE
Stream

### PR DESCRIPTION
**Pull request by Megha-Joshi**
Screenshot (168)

The feature included in this PR:
- Added single video steaming page
- And updated the UI

### CODE REVIEW

File: src/pages/Stream/stream.css [hunk 1]
```diff
@@ -0,0 +1,41 @@
+.video-cont{
+    padding: 15px;
+}
+
+.video-creator{
+    text-align: start;
+    padding: 1px 5px 5px 6px;
+    /* margin: 0px 5px 5px 5px 5px; */
```
Comment (HeyNitin) [inline src/pages/Stream/stream.css:8]: you can remove this commented code

**Review by HeyNitin — APPROVED** [review]
There's a single line of commented code, you can remove it
Looks good to merge to me

File: src/App.js [hunk 2]
```diff
@@ -18,6 +19,7 @@ function App() {
         {/* Public Routes */}
```
Comment (nakulsharma15) [inline src/App.js:19]: You can remove comments after they're no longer needed

File: src/pages/Stream/stream.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+const watchLaterHandler = ( token, video) => {
+    videoState.watchLater.some((item) => item._id === video._id) ?
+    removeItemFromWatchLaterVideos(video._id, token) : addItemToWatchLaterVideos(token,video)
+    }
+
+const currVideo = videos.find((video) => video._id === videoID)
+console.log("current video",currVideo);
+console.log(videoID);
```
Comment (nakulsharma15) [inline src/pages/Stream/stream.jsx:33]: You can remove console logs after they're no longer required.

File: src/context/videoContext.js [hunk 4]
```diff
@@ -99,12 +101,16 @@ const [ videoState, videoDispatch ] = useReducer(videoReducerFun, initialState);
...
-            videoDispatch({type: "SET_LIKED_VIDEOS", payload: response.likes})
-        }catch(error){
-            console.log(error);
+        if(token){
+            try{
+                console.log("entered like page");
+                const response = await addItemToLikedVideosHandler(token,video);
+                videoDispatch({type: "SET_LIKED_VIDEOS", payload: response.likes})
+            }catch(error){
+                console.log(error);
```
Comment (nakulsharma15) [inline src/context/videoContext.js:110]: You can add a toast here to show the error message to the user.

File: src/context/videoContext.js [hunk 5]
```diff
@@ -118,12 +124,16 @@ const [ videoState, videoDispatch ] = useReducer(videoReducerFun, initialState);
...
     const addItemToWatchLaterVideos = async (token,video) => {
-        try{
-            console.log("hello");
-            const response = await addItemToWatchLaterVideosHandler(token,video);
-            videoDispatch({type: "SET_WATCH_LATER",payload: response.watchlater});
-        }catch(error){
-            console.log(error);
+        if(token){
+            try{
+                console.log("hello");
```
Comment (nakulsharma15) [inline src/context/videoContext.js:129]: You can remove console.log after it's not needed

**Review by nakulsharma15 — COMMENTED** [review]
Suggested a few changes. Apart from that, everything looks good. I Liked how you named the functions and variables and I also liked the overall folder structure.

File: src/pages/Stream/stream.css (hunk 1 above)
Comment (Megha-Joshi) [reply src/pages/Stream/stream.css:8]: Okay

File: src/pages/Stream/stream.jsx (hunk 3 above)
Comment (Megha-Joshi) [reply src/pages/Stream/stream.jsx:33]: Okay

File: src/context/videoContext.js (hunk 4 above)
Comment (Megha-Joshi) [reply src/context/videoContext.js:110]: Okay

File: src/App.js (hunk 2 above)
Comment (Megha-Joshi) [reply src/App.js:19]: I used it for the reference.

**Pull request merged**